datahub-web-react/src/app/search/AdvancedSearchFilter.tsx

```diff
@@ -0,0 +1,126 @@
+import { CloseOutlined } from '@ant-design/icons';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import { FacetFilterInput, FacetMetadata } from '../../types.generated';
+import { ANTD_GRAY } from '../entity/shared/constants';
+import { AdvancedSearchFilterConditionSelect } from './AdvancedSearchFilterConditionSelect';
+import { SearchFilterLabel } from './SearchFilterLabel';
+import { AdvancedFilterSelectValueModal } from './AdvancedFilterSelectValueModal';
+import { FIELD_TO_LABEL } from './utils/constants';
+
+type Props = {
+    facet: FacetMetadata;
+    filter: FacetFilterInput;
+    onClose: () => void;
+    onUpdate: (newValue: FacetFilterInput) => void;
+};
+
+const FilterContainer = styled.div`
+    box-shadow: 0px 0px 4px 0px #00000010;
+    border-radius: 10px;
+    border: 1px solid ${ANTD_GRAY[4]};
+    padding: 5px;
+    margin: 4px;
+    :hover {
+        cursor: pointer;
+        background: ${ANTD_GRAY[2]};
+    }
+`;
+
+const FieldFilterSection = styled.span`
+    color: ${ANTD_GRAY[9]};
+    padding: 4px;
+    display: flex;
+    justify-content: space-between;
+`;
+
+const ValueFilterSection = styled.div`
+    :hover {
+        cursor: pointer;
+    }
+    border-top: 1px solid ${ANTD_GRAY[3]};
+`;
+
+const CloseSpan = styled.span`
+    :hover {
+        color: black;
+    }
+`;
+
+const StyledSearchFilterLabel = styled.div`
+    margin: 4px;
+`;
+
+const FilterFieldLabel = styled.span`
+    font-weight: 600;
+    margin-right: 2px;
+`;
+
+const TEXT_FILTERS = ['fieldPaths'];
+
+export const AdvancedSearchFilter = ({ facet, filter, onClose, onUpdate }: Props) => {
+    const [isEditing, setIsEditing] = useState(false);
+    return (
+        <>
+            <FilterContainer
+                onClick={() => {
+                    setIsEditing(!isEditing);
+                }}
+            >
+                <FieldFilterSection>
+                    <span>
+                        <FilterFieldLabel>{FIELD_TO_LABEL[filter.field]} </FilterFieldLabel>
+                        <AdvancedSearchFilterConditionSelect filter={filter} onUpdate={onUpdate} />
+                    </span>
+                    <CloseSpan
+                        role="button"
+                        onClick={(e) => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            onClose();
+                        }}
+                        tabIndex={0}
+                        onKeyPress={onClose}
+                    >
+                        <CloseOutlined />
+                    </CloseSpan>
+                </FieldFilterSection>
+                <ValueFilterSection>
+                    {TEXT_FILTERS.indexOf(filter.field) === -1 &&
```

Comment:
can we factor out this `TEXT_FILTERS.indexOf(filter.field) === -1` and rename it as a boolean variable? it's a little confusing to read, and we do use it multiple times. I imagine this just means that the filters don't contain that filter field

Reply:
also i think we should just use `.includes()`

Reply:
thanks for brining this up-- i realize we didn't need this logic at all